Delete output apphost upon failure of the CreateAppHost task.

peterhuene opened the pull request:
This commit deletes the output apphost when the `CreateAppHost` task fails from
an exception.

Partially fixes #2989.

Commit:
Delete output apphost upon failure of the CreateAppHost task.

This commit deletes the output apphost when the `CreateAppHost` task fails from
an exception.

Partially fixes #2989.

rainersigwald commented on src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs:
```diff
@@ -52,45 +52,54 @@ internal static class AppHost
+            catch (Exception)
+            {
+                // Delete the destination file so we don't leave an unmodified apphost
+                File.Delete(appHostDestinationFilePath);
```
nit: what if this throws, say because the current user doesn't have permission to write or delete the file?

peterhuene replied:
I'll catch the specific exceptions and log them if a logger was provided.

Commit:
Log an error if we fail to delete AppHost on error.

This commit logs an error message if we fail to delete the AppHost upon
failure.

peterhuene commented:
@rainersigwald requested change was pushed.
